ITE/4/README.adoc

```diff
@@ -0,0 +1,510 @@
+in-toto metadata as they're unaware of how to resolve certain URI tokens used by
+the other. This is again because of the possibility of inspections containing
+URIs that the other system is unable to resolve and calculate cryptographic
+information for.
+
+[[security]]
+== Security
+
+As per the in-toto specification, the only direct interaction in-toto tools have
+with artifacts is to record their hashes using one or more cryptographic hash
+algorithms. Other artifact operations such as verifying the artifact rules
+rely on the hashes recorded in the link metadata.
+
+We found that the recording of hashes of abstract resources, some of which may
+live at remote locations, is more complicated than the recording of hashes of
+artifacts in the local filesystem. It is, therefore, important for ITE-4
+compliant in-toto tools to handle the resolving of generic URIs to these
+abstract resources securely. It is quite possible that the contents of these
+resources need to be serialized in some manner that allows for their hashing,
+and these operations when performed unsafely can lead to severe vulnerabilities
+if the resource is controlled by a malicious actor.
+
+Further, it is likely that abstract resources change more frequently, both in
+content and format, and implementors must take care to identify how these
+contents are recorded, as well what specific information is recorded for a
+particular type of entity. Otherwise, the in-toto verification workflow can be
+plagued by failures due to the lack of availability of the artifact as
+previously recorded. For example, implementors who decide how to resolve GitHub
+pull request must decide what information encoded in a pull request must be
+hashed. If they are considering the comments of various users when hashing the
+pull request, this can lead to differences in the hashes recorded by different
+steps in the supply chain even if the changes proposed in the pull request are
+the same. This will of course eventually lead to a failure of the in-toto
+verification workflow.
+
+As always, it is also necessary to consider the actors who can make changes to
+an abstract entity. This is perhaps slightly exacerbated in the case of abstract
+resources as the content and format being hashed aren't as specific as artifacts
+in the local filesystem. In the above example, for a public repository, *any*
+GitHub user can comment on a pull request, so for an implementation that also
+considers comments when recording the hash, any user can potentially cause a
+failure of the verification workflow. This can potentially be leveraged by a
+malicious actor to target automated pipelines that rely on in-toto verification,
+using a Denial of Service attack.
+
+Finally, our analysis showed that verification of artifact rules specified in
+in-toto layouts use the hashes recorded while generating the link metadata and
+do not record afresh any of the hashes. So the security properties of artifact
```

Comment:
What does the word 'security' mean here?

Reply:
What I'm trying to convey is that we don't have anything to worry about just from verification of rules as long as all the previous points about resolving are implemented - being careful with serializing, changes in abstract resources, (now) outright missing resources, and being careful with repeatedly changing resources (such as comments on a PR). This will of course be further affected / limited depending on the other discussion about limiting scope to static resources. I may be using the wrong words here.

Reply:
Can you change the word in the text so that you are more specific?

Reply:
I rewrote this bit to read:

> Finally, our analysis showed that verification of artifact rules specified in in-toto layouts rely on the hashes recorded while generating link metadata for steps and inspections, and do not record any hashes afresh. Therefore, while the changes proposed in this ITE impact the resolving of artifacts and the recording of their hashes, they do not change how artifact rules are verified.